runtime/testy/stateManager.py

```diff
@@ -91,21 +98,40 @@ def setValue(self, value, keys):
     currDict = self.state
     i = 0
     try:
+      path = []
       for key in enumerate(keys[:-1]):
         i = key[0]
-        currDict = currDict[key[1]]
+        path.append(currDict[key[1]])
+        currDict = currDict[key[1]][0]
       if len(keys) > 1:
         i += 1
       if keys[i] not in currDict:
         raise Exception
-      currDict[keys[i]] = value
+      path.append(currDict[keys[i]])
+      currDict[keys[i]][0] = value
+      currTime = time.time();
+      for item in path:
+        item[1] = currTime
       self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(value)
     except:
       error = StudentAPIKeyError(self.dictErrorMessage(i, keys, currDict))
       self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(error)
 
+  def getTimestamp(self, keys):
+    currDict = self.state
+    timestamp = 0
+    try:
+      for key in enumerate(keys):
+        i = key[0]
+        timestamp = currDict[key[1]][1]
+        currDict = currDict[key[1]][0]
+      self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(timestamp)
+    except:
+      error = StudentAPIKeyError(self.dictErrorMessage(i, keys, result))
+      self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(error)
+
   def studentCodeTick(self):
-    self.state["runtime_meta"]["studentCode_main_count"] += 1
+    self.state["runtime_meta"][0]["studentCode_main_count"][0] += 1
```

Comment:
Ugh, this is getting ugly. ( Not your fault)
Let's start thinking of ways we might be able to get rid of this though in a future commit